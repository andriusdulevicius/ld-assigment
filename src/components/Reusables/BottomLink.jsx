import React from 'react';
import { ArrowRight } from '../../assets/icons/ArrowRight';
import { Typography, Link, Box } from '@mui/material';

const BottomLink = ({ linkColor, linkText, linkTo }) => {
  return (
    <Link color={linkColor} to={linkTo}>
      <Box display='flex' alignItems='center' flexWrap='nowrap' gap='0.3rem'>
        <Typography>{linkText}</Typography>
        <ArrowRight />
      </Box>
    </Link>
  );
};

export default BottomLink;
