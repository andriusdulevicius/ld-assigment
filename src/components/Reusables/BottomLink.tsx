import { FC } from 'react';
import { makeStyles } from '@mui/styles';
import { ArrowRight } from '../../assets/icons/ArrowRight';
import { Typography, Link, Box } from '@mui/material';

interface Props {
  text?: string;
  href?: string;
  color?: string;
}

const BottomLink: FC<Props> = ({ text, ...rest }) => {
  const classes = useStyles();

  return (
    <Link {...rest}>
      <Box className={classes.wrapper}>
        <Typography>{text}</Typography>
        <ArrowRight />
      </Box>
    </Link>
  );
};

export default BottomLink;

const useStyles = makeStyles({
  wrapper: {
    display: 'flex',
    alignItems: 'center',
    flexWrap: 'nowrap',
    gap: '.3rem',
    cursor: 'pointer',
  },
});
