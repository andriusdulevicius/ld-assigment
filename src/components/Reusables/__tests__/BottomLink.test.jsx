import { screen, render, cleanup } from '@testing-library/react';
import { mount } from 'jest';
import BottomLink from './BottomLink';
import renderer from 'react-test-renderer';

describe('ButtonLink', () => {
  afterEach(() => {
    cleanup();
  });

  it('should match snapshot', () => {
    const text = 'test link';
    const wrapper = mount(<BottomLink text={text} />);

    console.log(wrapper.debug());
    expect(wrapper.getByText('test link').exists()).toBeTruthy();

    expect().toBeDefined();
    expect().toBeUndefined();
    expect(wrapper.find('span')).toBe();
    expect().toEqaul();
    expect().toBeTruthy();
    expect().tobeFalsy();
    expect().toHaveBeenCalled();
  });
});
