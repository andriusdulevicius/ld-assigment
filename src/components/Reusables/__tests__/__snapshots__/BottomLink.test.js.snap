// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="makeStyles-wrapper-1 MuiBox-root css-0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      test link
    </p>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="undefinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.16667 10H15.8333"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          d="M10 4.16667L15.8333 10L10 15.8333"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </svg>
  </div>
</a>
`;
